front/pages/w/[wId]/ds/new.tsx

```diff
@@ -292,6 +294,33 @@ export default function DataSourceNew({
                     </div>
                   </div>
                 </div>
+                <div className="sm:col-span-6">
+                  <fieldset className="mt-2">
+                    <legend className="contents text-sm font-medium text-gray-700">
+                      Upload rights
+                    </legend>
+                    <div className="mt-1 flex items-center space-y-4">
+                      <input
+                        id="dataSourceUpsertable"
+                        name="upsertable"
+                        type="checkbox"
+                        className="h-4 w-4 cursor-pointer border-gray-300 text-violet-600 focus:ring-violet-500"
+                        checked={userUpsertable}
+                        onChange={(e) => setUserUpsertable(e.target.checked)}
+                      />
+                      <label
+                        htmlFor="upsertable"
+                        className="ml-3 block text-sm font-medium text-gray-700"
+                      >
+                        User-upsertable
```

Comment:
Let's call it "Users can upsert"

Reply:
Maybe we should say upload instead of upsert? less technical

Reply:
Definitely :+1: